lib/features/family/features/home_screen/cubit/navigation_bar_home_cubit.dart

```diff
@@ -1,12 +1,121 @@
+import 'dart:async';
+
+import 'package:collection/collection.dart';
+import 'package:givt_app/core/logging/logging.dart';
+import 'package:givt_app/features/auth/repositories/auth_repository.dart';
+import 'package:givt_app/features/children/add_member/models/member.dart';
+import 'package:givt_app/features/children/cached_members/repositories/cached_members_repository.dart';
 import 'package:givt_app/features/family/features/home_screen/presentation/models/navigation_bar_home_custom.dart';
+import 'package:givt_app/features/family/features/home_screen/presentation/models/navigation_bar_home_screen_uimodel.dart';
+import 'package:givt_app/features/family/features/home_screen/usecases/family_setup_usecase.dart';
+import 'package:givt_app/features/family/features/home_screen/usecases/preferred_church_usecase.dart';
+import 'package:givt_app/features/family/features/home_screen/usecases/registration_use_case.dart';
+import 'package:givt_app/features/family/features/profiles/models/profile.dart';
+import 'package:givt_app/features/family/features/profiles/repository/profiles_repository.dart';
+import 'package:givt_app/features/impact_groups/models/impact_group.dart';
+import 'package:givt_app/features/impact_groups/repo/impact_groups_repository.dart';
 import 'package:givt_app/shared/bloc/base_state.dart';
 import 'package:givt_app/shared/bloc/common_cubit.dart';
 
 class NavigationBarHomeCubit
-    extends CommonCubit<dynamic, NavigationBarHomeCustom> {
-  NavigationBarHomeCubit() : super(const BaseState.initial());
+    extends CommonCubit<NavigationBarHomeScreenUIModel, NavigationBarHomeCustom>
+    with PreferredChurchUseCase, RegistrationUseCase, FamilySetupUseCase {
+  NavigationBarHomeCubit(
+    this._profilesRepository,
+    this._authRepository,
+    this._impactGroupsRepository,
+    this._cachedMembersRepository,
+  ) : super(const BaseState.loading());
+
+  final ProfilesRepository _profilesRepository;
+  final AuthRepository _authRepository;
+  final ImpactGroupsRepository _impactGroupsRepository;
+  final CachedMembersRepository _cachedMembersRepository;
+
+  String? profilePictureUrl;
+  List<Profile> _profiles = [];
+  List<Member>? cachedMembers;
+  ImpactGroup? _familyInviteGroup;
+
+  Future<void> init() async {
+    _profilesRepository.onProfilesChanged().listen(_onProfilesChanged);
+    _cachedMembersRepository
```

Comment:
Should this also do a .fetchAllProfiles()? Or should I do it in an initState on the new style FamilyHomeScreen?

Reply:
At the moment it doesn't have to because other things are triggering the ProfilesRepository to refresh the data. The FamilyHomeScreen won't have to either. Profiles are already retrieved once you login which will trigger everyone who listens to a profiles update.